src/main/scala/org/broadinstitute/dsde/workbench/sam/openam/PostgresAccessPolicyDAO.scala

```diff
@@ -169,7 +169,7 @@ class PostgresAccessPolicyDAO(protected val dbRef: DbReference,
         (None, Option(nestedRoleRecord.role))
       }
     }.unzip
-    val descendantRoles = maybeDescendantRoles.toSet.flatten.groupBy(_._1).mapValues(rolesWithResourceTypeNames => rolesWithResourceTypeNames.map(_._2))
```

Comment:
new collections uses view to avoid copy pasting original collection for performance gain